carapace-server/src/main/java/org/carapaceproxy/client/impl/ConnectionsManagerImpl.java

```diff
@@ -86,9 +87,14 @@
             "stuck_requests_total",
             "stuck requests, this requests will be killed").register();
 
+    private static final int CONNECTIONS_MANAGER_THREADS_POOL_SIZE = Integer.getInteger("carapace.connectionsmanager.threadspool.size", 10);
```

Comment:
this is not good, as we know that this is used by commons pool to handle new connections

it is okay for an hotfix, but I expect this size to be computed from the number of maximum open connections